godot-macros/src/docs.rs

```diff
@@ -122,6 +123,26 @@ fn siphon_docs_from_attributes(doc: &[Attribute]) -> impl Iterator<Item = String
         })
 }
 
+fn xml_escape(value: &str) -> String {
+    let mut result = String::with_capacity(value.len());
+
+    for c in value.chars() {
+        if matches!(c, '&' | '<' | '>' | '"' | '\'') {
+            result.push_str(match c {
+                '&' => "&amp;",
+                '<' => "&lt;",
+                '>' => "&gt;",
+                '"' => "&quot;",
+                _ => "&#39;",
+            });
+        } else {
+            result.push(c)
+        }
```

Comment:
You said `if matches!` makes things 20% faster, but faster than what? In Release mode?

The following wouldn't need duplication:
```rs
        match c {
                '&' => result.push_str("&amp;"),
                '<' => result.push_str("&lt;"),
                '>' => result.push_str("&gt;"),
                '"' => result.push_str("&quot;"),
                '\'' => result.push_str("&#39;"),
                c => result.push(c),
            });
```

Reply:
>In Release mode?

Yes ofc

Your implementation was the very first thing I did, but then I added `matches!(...)` and noticed the improvement

However, today I tried to bench this again and noticed that the performance heavily depends on data (length and number of special characters), so maybe doing synthetic benchmarks doesn't even make sense here 🤔

Anyway, it's all about microseconds, so I can use your implementation not for performance but for simplicity

Reply:
Sounds good. I'm not against optimizations, but when they make code more complex or error-prone, we should have a good reason :slightly_smiling_face: